### PR TITLE
Adding TESTING.md

### PR DESCRIPTION
**Pull request by mcmellawatt**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [X] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [X] Is your initial contribution a single, squashed commit?

- [X] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: TESTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+`./gradlew project:testType --tests testName`
+For example:
+`./gradlew geode-core:test --tests ArrayUtilsTest`
+`./gradlew geode-core:distributedTest --tests ConnectionPoolDUnitTest`
+
+## Running Tests By Category
+To run a specific category of tests (eg: GfshTest):
+`./gradlew -PtestCategory=fullyQualifiedTestClassName testType`
+For example:
+`./gradlew -PtestCategory=org.apache.geode.test.junit.categories.GfshTest distributedTest`
```
Comment (moleske) [inline TESTING.md]: This example didn't work for me, but I'm feeling like maybe there's something I'm missing.

```
C:\Projects\geode [develop ≡]> ./gradlew.bat -PtestCategory=org.apache.geode.test.junit.categories.GfshTest distributedTest

FAILURE: Build failed with an exception.

* What went wrong:
Task '.apache.geode.test.junit.categories.GfshTest' not found in root project 'geode'.

* Try:
Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 7s
```

Maybe I was supposed to add the right project first?

File: TESTING.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Running Tests
+
+## Prerequisites
+Before running tests, ensure all steps in the [BUILDING.md](https://github.com/apache/geode/blob/develop/BUILDING.md) have been performed.
+
+## Running Tests By Type
```
Comment (moleske) [inline TESTING.md]: Just looking at the other checks on GitHub, do we not care about AcceptanceTest, StressNewTest, or UpgradeTest?  StressNewTest has zero tests from a quick glance and the other two have corresponding `gradle` calls

File: TESTING.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+* Integration tests
+  `./gradlew integrationTest --parallel`
+* Distributed tests
+  `./gradlew distributedTest --parallel`
+
+## Running Individual Tests
+To run an individual test, you can either
+1. Run the test in your [IDE](https://github.com/apache/geode/blob/develop/BUILDING.md#setting-up-intellij)
+2. Run the from terminal by specifying the sub-project and test type:
+`./gradlew project:testType --tests testName`
```
Comment (pdxrunner) [inline TESTING.md]: This needs to be formatted differently. The template and example commands display as a single run-on paragraph. Would be better to have each command displayed on a separate line. This is also true for the `Running Tests By Category` section

Comment (mcmellawatt) [reply TESTING.md]: Thanks Ken - will fix

File: TESTING.md (hunk 2 above)
Comment (mcmellawatt) [reply TESTING.md]: I think it makes sense to include acceptance and upgrade tests in the instructions.  I'll add them.  StressNewTest is really for CI - we "stress" brand new tests by running them many times to expose any unforeseen timing issues.  I don't think it belongs in these instructions.

File: TESTING.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+## Prerequisites
+Before running tests, ensure all steps in the [BUILDING.md](https://github.com/apache/geode/blob/develop/BUILDING.md) have been performed.
+
+## Running Tests By Type
+Tests are broken up into three types - unit, integration, and distributed.  They can be executed using the following commands from the Geode repository root directory:
+* Unit tests
+  `./gradlew test --parallel`
+* Integration tests
+  `./gradlew integrationTest --parallel`
```
Comment (pivotal-jbarrett) [inline TESTING.md]: You don't want to run integration or distributed tests in parallel without enabling the docker containers.

File: TESTING.md (hunk 1 above)
Comment (mcmellawatt) [reply TESTING.md]: I was able to reproduce on Windows and have found a working revised command.  Thanks!

File: TESTING.md (hunk 4 above)
Comment (pivotal-jbarrett) [reply TESTING.md]: No, its kind of a pain and you need a pretty beefy machine.

**Pull request merged**